execution/executor-types/src/parsed_transaction_output.rs

```diff
@@ -16,7 +16,13 @@ pub struct ParsedTransactionOutput {
 
 impl ParsedTransactionOutput {
     pub fn parse_reconfig_events(events: &[ContractEvent]) -> impl Iterator<Item = &ContractEvent> {
-        events.iter().filter(|e| *e.key() == *NEW_EPOCH_EVENT_KEY)
+        events.iter().filter(|e| {
+            if let ContractEvent::V1(v1) = e {
+                *v1.key() == *NEW_EPOCH_EVENT_KEY
+            } else {
+                false
+            }
```

Comment:
might be good to just add these acceors to the ContractEvent itself rather than having to always do this match statement nonsense...